### ISSUE TITLE
Generate HTML from the LaTeX source as well?

### ISSUE THREAD

**Issue by eterps** [guidoism/tex-oberon#1]
The PDF is look really good already, thanks for that monumental effort!

I was wondering if the LaTeX source can be converted to HTML as well at some point? 

And I don't mean 1 big HTML blob, but eventually something like this: https://doc.rust-lang.org/stable/book/

It would also enable people to link to specific sections in the book within online discussions.
Otherwise people need to keep referring to the blob like this:

```
It's an old question, see the "piece-chain" structure Prof With chose for text-with-"looks" (aka "style") [http://www.inf.ethz.ch/personal/wirth/ProjectOberon/PO.Syste...](http://www.inf.ethz.ch/personal/wirth/ProjectOberon/PO.System.pdf) section 5.2, "Text Management"
```

( https://hn.algolia.com/?dateRange=all&page=0&prefix=true&query=http%3A%2F%2Fwww.inf.ethz.ch%2Fpersonal%2Fwirth%2FProjectOberon%2F&sort=byDate&type=comment )

**Comment by guidoism** [comment]
Yes, definitely. My thought process is that the TeX will be the definitive "gold-standard", book quality, and I will convert it to epub and html using pandoc and maybe some manual tweaks. 

The TeX as is isn't particularly bad or messy but I'm learning how to make macros as I go that will make the markup a lot more semantic and thus a lot easier to convert to (nice) html.

As of a few minutes ago I've finished the hard part, which is to copy all of the text and figures from the PDFs and do basic typesetting. Oh man, that was a lot of work, I'm happy it's over with. Next up is going over the text with a fine-toothed comb and making sure it's accurate and beautiful. At some point during that process I might take a look at the conversion to HTML.

**Comment by eterps** [comment]
Sounds good, thank you.

I tried to go ahead and use pandoc to convert the TeX file to HTML, but it couldn't process the TeX file as-is. Not sure if the TeX file needs some macro expansion first or something like that before pandoc can take it as plain LaTeX?
I am not too familiar with [La|Lua]Tex, so I can't tell.

Anyway don't bother yourself with that, you have plenty of work as you already indicated. I'll see if I can find some time to go over the resulting (PDF) text myself. For example I spotted that the proper indentation of the code fragments is a manual effort as well.

**Comment by guidoism** [comment]
It's actually plain TeX, not LaTeX, and it uses Knuth's macros for The Art of Computer Programming. I know pandoc knows about plain TeX at some point, but I can't remember if it's only for output or it knows how to use it for input too.

**Comment by Andonome** [comment]
Only TeX reads TeX. I've lost a few afternoons trying to translate it to anything else.

LaTeX now has a pretty mature `markdown` package, so if it began in markdown, the rest might work.

Alternatively, there's [typst](https://typst.app).

**Comment by eterps** [comment]
Also see my conversion attempt here: https://github.com/guidoism/tex-oberon/pull/4
